app/src/main/java/org/simple/clinic/home/overdue/search/OverdueSearchEffectHandler.kt

```diff
@@ -51,9 +51,22 @@ class OverdueSearchEffectHandler @AssistedInject constructor(
         .addConsumer(ClearSelectedOverdueAppointments::class.java, ::clearSelectedOverdueAppointments)
         .addTransformer(ReplaceSelectedAppointmentIds::class.java, replaceSelectedAppointmentIds())
         .addConsumer(ScheduleDownload::class.java, ::scheduleDownload)
+        .addTransformer(AddAllAppointmentIds::class.java, addAllAppointmentIds())
         .build()
   }
 
+  private fun addAllAppointmentIds(): ObservableTransformer<AddAllAppointmentIds, OverdueSearchEvent>? {
+    return ObservableTransformer { effects ->
+      effects
+          .observeOn(schedulersProvider.computation())
+          .map { (appointmentIds) ->
+            overdueAppointmentSelector.addSelectedIds(appointmentIds)
+            return@map appointmentIds
+          }
+          .map(::SelectedOverdueAppointmentsLoaded)
```

Comment:
We already have an observable stream from the selector to update model. So, I don't think we need to return those appointments back the `Update` function.